src/api.js

```diff
@@ -0,0 +1,139 @@
+
+export function cleanRecord(record, excludeFields=RECORD_FIELDS_TO_CLEAN) {
+  return Object.keys(record).reduce((acc, key) => {
+    if (excludeFields.indexOf(key) === -1)
+      acc[key] = record[key];
+    return acc;
+  }, {});
+};
+
+const DEFAULT_REQUEST_HEADERS = {
+  "Accept":       "application/json",
+  "Content-Type": "application/json",
+};
+
+export default class Api {
+  constructor(remote, options={}) {
+    if (typeof(remote) !== "string" || !remote.length)
+      throw new Error("Invalid remote URL: " + remote);
+    this.remote = remote;
+    try {
+      this.version = "v" + remote.match(/\/v(\d+)\/?$/)[1];
+    } catch (err) {
+      throw new Error("The remote URL must contain the version: " + remote);
+    }
+  }
+
+  endpoints(options={full: true}) {
+    var root = options.full ? this.remote : `/${this.version}`;
+    return {
+      root:           () => root,
+      batch:          () => `${root}/batch`,
+      collection: (coll) => `${root}/collections/${coll}/records`,
+      record: (coll, id) => `${this.endpoints(options).collection(coll)}/${id}`,
+    };
+  }
+
+  /**
+   * Fetches latest changes from the remote server.
+   *
+   * @param  {String} collName     The collection name.
+   * @param  {Number} lastModified Latest sync timestamp.
+   * @param  {Object} options      Options.
+   * @return {Promise}
+   */
+  fetchChangesSince(collName, lastModified=null, options={headers: {}}) {
+    var newLastModified;
+    var queryString = "?" + (lastModified ? "_since=" + lastModified : "");
+    return fetch(this.endpoints().collection(collName) + queryString, {
+      headers: Object.assign({}, DEFAULT_REQUEST_HEADERS, {
+        "If-Modified-Since": lastModified ? String(lastModified) : "0"
+      }, options.headers)
+    })
+      .then(res => {
+        // If HTTP 304, nothing has changed
+        if (res.status === 304) {
+          newLastModified = lastModified;
+          return {items: []};
+        } else if (res.status >= 400) {
+          // TODO: attach better error reporting
+          throw new Error("Fetching changes failed: HTTP " + res.status);
+        } else {
+          newLastModified = res.headers.get("Last-Modified");
+          return res.json();
+        }
+      })
+      .then(json => {
+        return {
+          lastModified: newLastModified,
+          changes: json.items
+        };
+      });
+  }
+
+  /**
+   * Sends batch update requests to the remote server.
+   *
+   * @param  {String} collName The collection name.
+   * @param  {Array}  records  The list of record updates to send.
+   * @param  {Object} headers  Headers to attach to each update request.
+   * @param  {Object} options  Options.
+   * @return {Promise}
+   */
+  batch(collName, records, headers={}, options={safe: true}) {
+    const results = {
+      errors:    [],
+      published: [],
+      conflicts: [],
+      skipped:   []
+    };
+    if (!records.length)
+      return Promise.resolve(results);
```

Comment:
Note for later: if more than X results (default is 25 on server), split in several calls.